app/uk/gov/hmrc/gform/builder/BuilderController.scala

```diff
@@ -68,7 +68,26 @@ class BuilderController(
   }
 
   private val compatibilityVersion =
-    14; // This is managed manually. Increase it any time API used by builder extension is changed.
+    15; // This is managed manually. Increase it any time API used by builder extension is changed.
+
+  def hiddenChoiceIndexes(formModelOptics: FormModelOptics[DataOrigin.Mongo]): Map[String, List[Int]] = {
+    val visibilityFormComponents = formModelOptics.formModelVisibilityOptics.formModel.allFormComponents
```

Comment:
Can this take into consideration only relevant components? We need to process only components for given `sectionNumber: SectionNumber`

```suggestion
    val visibilityFormComponents = formModelOptics.formModelVisibilityOptics.formModel(sectionNumber).allFormComponents
```